x-pack/plugins/maps/server/content_management/maps_storage.ts

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import Boom from '@hapi/boom';
+import type { ContentStorage, StorageContext } from '@kbn/content-management-plugin/server';
+import type { SavedObject, SavedObjectReference } from '@kbn/core-saved-objects-api-server';
+
+import { CONTENT_ID } from '../../common/content_management';
+import { cmServicesDefinition } from '../../common/content_management/cm_services';
+import type {
+  MapSavedObject,
+  PartialMapSavedObject,
+  MapContentType,
+  MapSavedObjectAttributes,
+  MapGetOut,
+  MapCreateIn,
+  MapCreateOut,
+  CreateOptions,
+  MapUpdateIn,
+  MapUpdateOut,
+  UpdateOptions,
+  MapDeleteOut,
+  MapSearchOut,
+  MapSearchQuery,
+} from '../../common/content_management';
+
+const savedObjectClientFromRequest = async (ctx: StorageContext) => {
+  if (!ctx.requestHandlerContext) {
+    throw new Error('Storage context.requestHandlerContext missing.');
+  }
+
+  const { savedObjects } = await ctx.requestHandlerContext.core;
+  return savedObjects.client;
+};
+
+type PartialSavedObject<T> = Omit<SavedObject<Partial<T>>, 'references'> & {
+  references: SavedObjectReference[] | undefined;
+};
+
+function savedObjectToMapItem(
+  savedObject: SavedObject<MapSavedObjectAttributes>,
+  partial: false
+): MapSavedObject;
+
+function savedObjectToMapItem(
+  savedObject: PartialSavedObject<MapSavedObjectAttributes>,
+  partial: true
+): PartialMapSavedObject;
+
+function savedObjectToMapItem(
```

Comment:
This has me a bit confused. As best I can tell its changing some object keys from snake case to camel case.

Reply:
It is a function that guarantees that we explicitly return the fields for our `MapItem` instead of doing `{ attributes: ...savedObject.attributes }`